Mage/src/main/java/mage/abilities/common/CastCommanderAbility.java

```diff
@@ -1,38 +0,0 @@
-package mage.abilities.common;
-
-import mage.abilities.SpellAbility;
-import mage.cards.Card;
-import mage.constants.Zone;
-
-/**
- * @author Plopman, JayDi85
- */
-public class CastCommanderAbility extends SpellAbility {
-
-    private final String ruleText;
-
-    public CastCommanderAbility(Card card, SpellAbility spellTemplate) {
-        super(spellTemplate);
-        this.newId();
-        this.setCardName(spellTemplate.getCardName());
-        this.zone = Zone.COMMAND;
-        this.spellAbilityType = spellTemplate.getSpellAbilityType();
-        this.ruleText = spellTemplate.getRule(); // need to support custom rule texts like OverloadAbility
```

Comment:
What about that comment about custom rules for OverloadAbility? Is it works or broken with new changes?

Reply:
It works now since the new `SpellAbility` would still be an `OverloadAbility`, rather than becoming a generic `CastCommanderAbility`. This is the fundamental problem that this PR solves, as the `SpellAbility` for casting transformed affects more than just the custom rules text.